scanpointgenerator/core/compoundgenerator.py

```diff
@@ -76,36 +77,47 @@ def prepare(self):
         # we should restrict the resulting grid rather than merge dimensions
         # this changes the alternating case a little (without doing this, we
         # may have started in reverse direction)
-        for rect in [r for r in excluders \
-                if isinstance(r.roi, RectangularROI) and r.roi.angle == 0]:
-            axis_1, axis_2 = rect.scannables[0], rect.scannables[1]
-            gen_1 = [g for g in generators if axis_1 in g.axes][0]
-            gen_2 = [g for g in generators if axis_2 in g.axes][0]
-            if gen_1 is gen_2:
-                continue
-            if isinstance(gen_1, LineGenerator) \
-                    and isinstance(gen_2, LineGenerator):
-                gen_1.prepare_positions()
-                gen_2.prepare_positions()
-                valid = np.full(gen_1.size, True, dtype=np.int8)
-                valid &= gen_1.positions[axis_1] \
-                        <= rect.roi.width + rect.roi.start[0]
-                valid &= gen_1.positions[axis_1] >= rect.roi.start[0]
-                points_1 = gen_1.positions[axis_1][valid.astype(np.bool)]
-                valid = np.full(gen_2.size, True, dtype=np.int8)
-                valid &= gen_2.positions[axis_2] \
-                        <= rect.roi.height + rect.roi.start[1]
-                valid &= gen_2.positions[axis_2] >= rect.roi.start[1]
-                points_2 = gen_2.positions[axis_2][valid.astype(np.bool)]
-                new_gen1 = LineGenerator(
-                    gen_1.axes, gen_1.units, points_1[0], points_1[-1],
-                    len(points_1), gen_1.alternate)
-                new_gen2 = LineGenerator(
-                    gen_2.axes, gen_2.units, points_2[0], points_2[-1],
-                    len(points_2), gen_2.alternate)
-                generators[generators.index(gen_1)] = new_gen1
-                generators[generators.index(gen_2)] = new_gen2
-                excluders.remove(rect)
+        for excluder_ in [e for e in excluders if isinstance(e, ROIExcluder)]:
+            rect_rois = [r for r in excluder_.rois
+                         if isinstance(r, RectangularROI) and r.angle == 0]
+            if len(rect_rois) == 1:
```

Comment:
No, because you'd potentially be excluding points contained in the other region within the same excluder, when we're supposed to be considering their *union*.